### ISSUE TITLE
Valid Webhook URL being rejected by pattern match

### ISSUE THREAD

**Issue by Flash-Git** [0x4e49434f4c45/amcdb#20]
I'm getting the ` does not appear to be a valid Discord webhook URL! Webhook mode will not be enabled.` message no matter how many urls I generate and try in the config.

**Comment by 0x4e49434f4c45** [comment]
I was worried that might happen as I only had mine to go off of.

Can you generate a few that have this problem and post them here? Just make sure to delete them from Discord first so nobody can use them.

**Comment by Flash-Git** [comment]
Sure thing:

Failing:
1. `https://discord.com/api/webhooks/1081958472469446687/cutxroAo_H3Nb_HUEnaOOb9T2Tjdy8y0naKUry4Di9n4pZGapW8YUDW0Y-n3XrVQrQr6`
2. `https://discord.com/api/webhooks/1081972586306609182/v03xn_ErOe5BeLEAh2JT4eckNTOHOylA-pmTsag0hNfnXuOyVSBkHem0BNzBFVgokF2W`
3. `https://discord.com/api/webhooks/1081972896437653646/0j0MHJwOKfmWkIPTULNR9wkXh-Yrwrp9GPR1gGk1Ms2QYVCvgnMDEH-tODWgj0-cKv3d`

Funny enough my 4th test (7th webhook generated) actually worked so I'm personally no longer in need of a fix, culprit looks like it could be the `-`.

**Comment by 0x4e49434f4c45** [comment]
What's actually happening there is it's falling back on posting directly to the configured chat channel via the bot.

At least that's what it's *supposed* to do, are you sure it's using the webhook?

**Comment by Flash-Git** [comment]
Yep definitely using the webhook and posting to chat when the url doesn't contain a dash, even when the configured chat channel option is disabled. It's how I knew that url was different, I wasn't getting notifications up until that point.

When the webhook was failing, it couldn't post to chat because that option was disabled.

**Comment by 0x4e49434f4c45** [comment]
Oh I misread your comment, I thought you were saying that when the webhook URL is rejected it goes ahead and posts via the webhook anyway.

The behavior of posting via the webhook if one is configured regardless of the chat channel configuration is intended -- I figured one wouldn't configure a webhook if they didn't want it to be used. I could add some clarification around that. Can you see a scenario where this would cause a problem?

**Comment by Flash-Git** [comment]
It's definitely niche, I think it could cause unexpected behaviour when performing debugging that requires multiple restarts.

e.g. I disabled the chat channel config to avoid spamming the channel with startup and shutdown messages while I restarted over and over trying to generate urls for you.

If I was doing that for another reason I would want to make sure to disable both the chat channel AND the webhook url. Not a problem and I personally won't forget!
